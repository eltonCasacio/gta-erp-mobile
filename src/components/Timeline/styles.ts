import styled, {css} from 'styled-components/native';
import Theme from '../../styles/Theme';

export const Wrapper = styled.View`
  background-color: ${Theme.Colors.white};
  align-items: center;

  padding: 0 ${Theme.Spacings.great};
  min-height: 150px;
`;

export type TextProps = {
  checked: boolean;
};

export const Text = styled.Text<TextProps>`
  ${({checked}) => css`
    color: ${checked ? Theme.Colors.grayBold : Theme.Colors.gray};
    font-size: ${Theme.Font.medium};
  `}
`;

export const TextWrapper = styled.View`
  flex-direction: row;
  justify-content: space-between;
  margin: ${Theme.Spacings.medium} 0;

  width: 100%;
`;
