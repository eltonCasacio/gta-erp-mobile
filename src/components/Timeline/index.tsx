import React from 'react';
import * as S from './styles';

const Timeline = ({apontamentos = []}) => {
  const apontamentoEl = apontamentos.map((item, indice) => {
    const {registerDate, registerHour, tipo} = item;

    return (
      <S.TextWrapper key={indice}>
        <S.Text checked>{tipo}</S.Text>
        <S.Text checked>{`${registerDate}   -   ${registerHour}`}</S.Text>
      </S.TextWrapper>
    );
  });

  return <S.Wrapper>{apontamentoEl}</S.Wrapper>;
};

export default Timeline;
