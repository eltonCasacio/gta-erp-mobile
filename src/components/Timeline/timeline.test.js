import React from 'react';
import renderer from 'react-test-renderer';

import Timeline from '.';

describe('<Timeline />', () => {
  it('should render default styles', () => {
    const tree = renderer.create(<Timeline />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
