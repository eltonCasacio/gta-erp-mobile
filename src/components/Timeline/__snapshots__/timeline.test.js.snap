// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timeline /> should render default styles 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fafafa",
        "paddingBottom": 0,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 5,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 5,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      checked={true}
      style={
        Array [
          Object {
            "color": "#707070",
            "fontSize": 16,
          },
        ]
      }
    >
      ENTRADA:
    </Text>
    <Text
      checked={true}
      style={
        Array [
          Object {
            "color": "#707070",
            "fontSize": 16,
          },
        ]
      }
    >
      06:00
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 5,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 5,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      checked={false}
      style={
        Array [
          Object {
            "color": "#AEAEAE",
            "fontSize": 16,
          },
        ]
      }
    >
      SAÍDA:
    </Text>
    <Text
      checked={false}
      style={
        Array [
          Object {
            "color": "#AEAEAE",
            "fontSize": 16,
          },
        ]
      }
    >
      10:00
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 5,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 5,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      checked={false}
      style={
        Array [
          Object {
            "color": "#AEAEAE",
            "fontSize": 16,
          },
        ]
      }
    >
      ENTRADA:
    </Text>
    <Text
      checked={false}
      style={
        Array [
          Object {
            "color": "#AEAEAE",
            "fontSize": 16,
          },
        ]
      }
    >
      12:00
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 5,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 5,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      checked={false}
      style={
        Array [
          Object {
            "color": "#AEAEAE",
            "fontSize": 16,
          },
        ]
      }
    >
      SAÍDA:
    </Text>
    <Text
      checked={false}
      style={
        Array [
          Object {
            "color": "#AEAEAE",
            "fontSize": 16,
          },
        ]
      }
    >
      18:00
    </Text>
  </View>
</View>
`;
